tests/ixia/test_ixia_traffic_restpy.py

```diff
@@ -0,0 +1,183 @@
+
+import logging
+import time
+import pytest
+import pprint
+from common.utilities import wait_until
+from common.fixtures.conn_graph_facts import conn_graph_facts
+from common.platform.interface_utils import check_interface_information
+from common.platform.daemon_utils import check_pmon_daemon_status
+from common.reboot import *
+from common.platform.device_utils import fanout_switch_port_lookup
+from common.helpers import assertions
+from ixnetwork_restpy import SessionAssistant, Files
+
+from common.ixia_fixtures import ixia_api_serv_ip, ixia_api_serv_user,\
+     ixia_api_serv_passwd, ixia_dev, ixia_api_serv_port, ixia_api_serv_session_id
+
+from common.ixia_helpers import get_neigh_ixia_mgmt_ip, get_neigh_ixia_card,\
+     get_neigh_ixia_port
+
+import time
+
+#pytestmark = [pytest.mark.disable_loganalyzer]
+
+def returnIxiaChassisIp (chassisDict, no) :
+    chList = []
+    for key in chassisDict.keys():
+        chList.append(chassisDict[key])
+
+    return (chList[no - 1])
+
+def parseDeviceConn (device_conn) :
+    retval = []
+    dive_con_dict = device_conn['device_conn']
+    for key in dive_con_dict.keys() :
+        pp =  dive_con_dict[key]['peerport']
+        string = pp + '/' + key
+        retval.append(string)
+    retval.sort()
+    return(retval)
+
+def getCard(ixiaCardPortList, num) :
+    card = ixiaCardPortList[num].split('/')[0]
+    cardNo = int(card.replace('Card', ''))
+    return(cardNo)
+
+def getPort(ixiaCardPortList, num) :
+    port = ixiaCardPortList[num].split('/')[1]
+    portNo = int(port.replace('Port', ''))
+    return(portNo)
+
+def test_testbed(testbed, conn_graph_facts, duthost, ixia_dev, ixia_api_serv_ip,
+                 ixia_api_serv_user, ixia_api_serv_passwd, ixia_api_serv_port,
+                 ixia_api_serv_session_id):
+    logger.info("Connection Graph Facts = %s " %(conn_graph_facts))
+    logger.info("DUT hostname = %s" %(duthost.hostname))
+    ixiaports = parseDeviceConn(conn_graph_facts)
+    logger.info("Ixia ports = %s" %(ixiaports))
+    logger.info("Ixia chassis IP = %s" %(ixia_dev))
+    logger.info("Ixia API server IP = %s" %(ixia_api_serv_ip))
+    logger.info("Ixia API server user = %s" %(ixia_api_serv_user))
+    logger.info("Ixia API server password = %s" %(ixia_api_serv_passwd))
+    logger.info("Ixia API server port = %s" %(ixia_api_serv_port))
+    logger.info("Ixia API server sessionId = %s" %(ixia_api_serv_session_id))
+
+    clientIp  = ixia_api_serv_ip
+    UserName  = ixia_api_serv_user
+    Password  = ixia_api_serv_passwd
+    RestPort  = ixia_api_serv_passwd
+    SessionId = ixia_api_serv_session_id
+    chassisIp = returnIxiaChassisIp(ixia_dev, 1)
+
+    cardId    = getCard(ixiaports, 0)
+    PortId1   = getPort(ixiaports, 0)
+    PortId2   = getPort(ixiaports, 1)
+    PortId3   = getPort(ixiaports, 2)
+    PortId4   = getPort(ixiaports, 3)
+    PortId5   = getPort(ixiaports, 4)
+    PortId6   = getPort(ixiaports, 5)
```

Comment:
Unless 6 ixia connections exist, this test cannot be used by others, can you make this num_of_ports configurable?

Reply:
All this hard codings are removed. I have used  "IxiaFanoutManager" class which can handle variable number of physical ports (as comes in fanout_graph_fact) and creates that number of virtual ports in the ixNetwork configuration. Also port assignments are done accordingly.